File: generated-sources/kotlin-spring/mojang-sessions/src/main/kotlin/org/openapitools/model/PlayerSkinURL.kt
package org.openapitools.model

import java.util.Objects
import org.openapitools.model.PlayerTextureURL
import javax.validation.Valid
import javax.validation.constraints.*

/**
 * Wraps the URL to the texture and configures the player's model
 */
data class PlayerSkinURL (

        @get:NotNull 
        @JsonProperty("url") override val url: String
) : PlayerTextureURL{

}

File: generated-sources/scala-finch/mojang-sessions/src/main/scala/org/openapitools/models/PlayerSkinURL.scala
package org.openapitools.models

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import org.openapitools._
import org.openapitools.models.PlayerTextureURL

/**
 * Wraps the URL to the texture and configures the player's model
 * @param url The URL to the texture, must be in Mojang's domains.
 */
case class PlayerSkinURL(url: String
                )

object PlayerSkinURL {
    /**
     * Creates the codec for converting PlayerSkinURL from and to JSON.
     */
    implicit val decoder: Decoder[PlayerSkinURL] = deriveDecoder
    implicit val encoder: ObjectEncoder[PlayerSkinURL] = deriveEncoder
}
